StringOperations.java

```diff
@@ -0,0 +1,199 @@
+/***
+ * To ask the user which operation they want to execute on the string. The available operations are
+ * listed below: Append, CountWords, Replace, isPalindrome, Splice, Split,
+ * MaxRepeatingCharacter, Sort, Shift, and Reverse.
+ */
+package Assignment_2;
+
+import java.util.*;
+
+public class StringOperations {
+
+    Scanner input = new Scanner(System.in);
+
+    final public void Append()
+    {
+        Constant constant = new Constant();
+        char choice;
+
+        System.out.println(constant.FIRST_STRING);
+        String str = input.nextLine();
+
+        do {
+            System.out.println(constant.ANOTHER_STRING);
+            String str2 = input.next();
+            str = str + " " + str2;
+            System.out.println(str);
+            System.out.println(constant.APPEND_STRING);
+            choice = input.next().charAt(0);
+        }while(choice == 'y' || choice == 'Y');
+
+    }
+
+    final public void WordsCount() {
+
+        Constant constant = new Constant();
+        System.out.println(constant.ENTER_STRING);
+        String str = input.nextLine();
+
+    }
+
+    final public void Replace() {
+
+        Constant constant = new Constant();
+        System.out.println(constant.ENTER_STRING);
+        String str = input.nextLine();
+        int str_len = str.length();
+        System.out.println(constant.PREV_STRING);
+        String x = input.nextLine();
+        int x_length = x.length();
+        System.out.println(constant.REPLACE_STRING);
+        String v = input.nextLine();
+        String new_string = "";
+        for( int i = 0; i < str.length(); i++ )
+        {
+            if(str.charAt(i) == x.charAt(0) )
+            {
+                new_string = str.substring(0,i-1) + v + str.substring(x_length+1,str_len);

```

Comment:
add comment to that this is or optimize the condition with good variables to have a better understanding